Sources/Tracing/Docs.docc/InDepthGuide.md

```diff
@@ -0,0 +1,332 @@
+   users becomes `context.baggage` (rather than the confusing `context.context`)
+2. If you cannot or it would not make sense to carry baggage inside your framework's context object, pass (and accept (!)) the `LoggingContext` in your framework functions like follows:
+- if they take no framework context, accept a `context: LoggingContext` which is the same guideline as for all other cases
+- if they already _must_ take a context object and you are out of words (or your API already accepts your framework context as "context"), pass the baggage as **last** parameter (see above) yet call the parameter `baggage` to disambiguate your `context` object from the `baggage` context object.
+
+Examples:
+
+- `Lamda.Context` may contain `baggage` and a `logger` and should be able to conform to `LoggingContext`
+  - passing context to a `Lambda.Context` unaware library becomes: `http.request(url: "...", context: context)`.
+- `ChannelHandlerContext` offers a way to set/get baggage on the underlying channel via `context.baggage = ...`
+  - this context is not passed outside a handler, but within it may be passed as is, and the baggage may be accessed on it directly through it.
+  - Example: [https://github.com/apple/swift-nio/pull/1574](https://github.com/apple/swift-nio/pull/1574)
+
+### Creating context objects (and when not to do so)
+
+Generally application developers _should not_ create new context objects, but rather keep passing on a context value that they were given by e.g. the web framework invoking the their code.
+
+If really necessary, or for the purposes of testing, one can create a baggage or context using one of the two factory functions:
+
+- [`DefaultLoggingContext.topLevel(logger:)`](https://github.com/apple/swift-distributed-tracing-baggage/blob/main/Sources/Baggage/LoggingContext.swift#L232-L259) or [`Baggage.topLevel`](https://github.com/apple/swift-distributed-tracing-baggage-core/blob/main/Sources/CoreBaggage/Baggage.swift#L79-L103) - which creates an empty context/baggage, without any values. It should _not_ be used too frequently, and as the name implies in applications it only should be used on the "top level" of the application, or at the beginning of a contextless (e.g. timer triggered) event processing.
+- [`DefaultLoggingContext.TODO(logger:reason:)`](https://github.com/apple/swift-distributed-tracing-baggage/blob/main/Sources/Baggage/LoggingContext.swift#L262-L292) or [`Baggage.TODO`](https://github.com/apple/swift-distributed-tracing-baggage-core/blob/main/Sources/CoreBaggage/Baggage.swift#L107-L136) - which should be used to mark a parameter where "before this code goes into production, a real context should be passed instead." An application can be run with `-DBAGGAGE_CRASH_TODOS` to cause the application to crash whenever a TODO context is still in use somewhere, making it easy to diagnose and avoid breaking context propagation by accidentally leaving in a `TODO` context in production.
```

Comment:
Same here

Reply:
Heh, I missed those :( I guess we should not link at all -- or perhaps just to the file itself?

docc should really fix their multi package story, we don't have to work around it too hard.

Reply:
I've changed these to link to the files only.